### PR TITLE
AI Assistant: Fix crash when transforming to AI Assistant block

### LINKED ISSUE #31691: AI Extension: editor crash when using Ask AI on two paragraphs
Reproduced on Jetpack 12.3-beta and bleeding edge at 7bc77d1

**To reproduce:**

* With AI features enabled, in the post editor, add two separate paragraph blocks.
* Select both paragraph blocks, then click on the AI Assistant block toolbar.
* Click on "Ask AI Assistant"
* Note the second paragraph block is duplicated.
* Click on "Accept" for the Jetpack AI suggestion
* Editor hard crash with the following error:

```
TypeError: Cannot read properties of undefined (reading 'right')
    at Object.getBoundingClientRect (index.js:134:22) . . . 
```

Reported via internal testing: p8oabR-1eP-p2#comment-7423

### PR DESCRIPTION
**Pull request by renatoagds**
Closes #31691 

## Proposed changes:

* Add `@wordpress/block-editor` types.
* Move `getTextContentFromSelectedBlocks ` to be an hook (`useTextContentFromSelectedBlocks `).
* Remove unused blocks when transforming to `AI Assistant`.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

N/A

## Testing instructions:

* Add paragraphs in your post.
* Select multiple (2, 3, 4 ...).
* Convert them using `Ask Ai Assistant` button.
* They should be all converted and current blocks removed.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/utils/block-content.ts [hunk 1]
```diff
@@ -63,8 +64,23 @@ type GetTextContentFromBlocksProps = {
  * @returns {GetTextContentFromBlocksProps} The text content.
  */
 
-export function getTextContentFromSelectedBlocks(): GetTextContentFromBlocksProps {
-	const clientIds = select( 'core/block-editor' ).getSelectedBlockClientIds();
+export function useTextContentFromSelectedBlocks(): GetTextContentFromBlocksProps {
```
Comment (retrofox) [inline projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/utils/block-content.ts]: could you move the hook to the hooks folder?

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/types.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import type * as BlockEditorSelectors from '@wordpress/block-editor/store/selectors';
```
Comment (retrofox) [inline projects/plugins/jetpack/extensions/blocks/ai-assistant/types.ts:1]: This is nice 💯

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/utils/block-content.ts (hunk 1 above)
Comment (renatoagds) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/utils/block-content.ts]: Sure. Doing it.

Comment (renatoagds) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/utils/block-content.ts]: Done [7e21505](https://github.com/Automattic/jetpack/pull/31813/commits/7e21505864ec3d315174986e5e6c24d85fa7ae07)

**Review by retrofox — APPROVED** [review]
Looks good to me! It resolves the problem from the root. 🚀

**Pull request merged**